Backend/lib/location-updates/src/Lib/LocationUpdates/Internal.hs

```diff
@@ -168,17 +171,18 @@ recalcDistanceBatches h@RideInterpolationHandler {..} ending driverId estDist es
       sendTollCrossedNotificationToDriver driverId
       sendTollCrossedUpdateToBAP driverId
   let recomputeIfPickupDropNotOutsideOfThreshold = getRecomputeIfPickupDropNotOutsideOfThreshold
-  let snapToRoadCallCondition = (routeDeviation && recomputeIfPickupDropNotOutsideOfThreshold) || pickupDropOutsideThreshold || isJust rectifyDistantPointsFailureUsing || isJust estTollCharges || tollRouteDeviation
+  let snapToRoadCallCondition = (routeDeviation && recomputeIfPickupDropNotOutsideOfThreshold) || pickupDropOutsideThreshold || isJust rectifyDistantPointsFailureUsing || isJust estTollCharges || tollRouteDeviation || passedThroughDrop
   if ending
     then do
       if snapToRoadCallCondition
         then do
           currSnapToRoadState <- processSnapToRoadCall
-          updateDistance driverId currSnapToRoadState.distanceTravelled currSnapToRoadState.googleSnapToRoadCalls currSnapToRoadState.osrmSnapToRoadCalls currSnapToRoadState.numberOfSelfTuned calculationFailed
-          when isTollApplicable $ do
-            mbTollCharges :: Maybe HighPrecMoney <- Redis.safeGet (onRideTollChargesKey driverId)
-            tollNames :: [Text] <- Redis.lRange (onRideTollNamesKey driverId) 0 (-1)
-            whenJust mbTollCharges $ \tollCharges -> updateTollChargesAndNames driverId tollCharges tollNames
+          when (not currSnapToRoadState.passThroughDropThreshold) $ do
```

Comment:
let's not stop the distance accumulation, instead we can update this flag in DB and during endRide use this flag to decide whether to recompute or not during accurate and inaccurate distance calculation cases.